harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -863,19 +881,8 @@ def get_trial_list_after_successful_run(
         self,
         last_trial: DSATTrial,
     ) -> List[DSATTrial]:
-        # TODO: remove below print tests.
-        logging.info("**************** BSZ History ****************")
-        bsz_history = []
-        print_trial = last_trial
-        while print_trial is not None:
-            bsz = print_trial.ds_config["train_micro_batch_size_per_gpu"]
-            bsz_history.append(bsz)
-            print_trial = print_trial.parent
-        logging.info(f"History (to-be-submitted last): {str(list(reversed(bsz_history)))}")
-        logging.info("**************** BSZ History End ****************")
-
         # TODO: verify we are always quitting when no more non-trivial trials are possible.
-        if self.should_stop_lineage(trial=last_trial):
+        if self.should_stop_lineage(trial=last_trial) or last_trial.search_data is None:
```

Comment:
When would `last_trial.search_data is None` happen? Seems like it shouldn't occur.

Reply:
Just according to our `DSATTrial` class, its `Optional`. If we can make sure that in all cases it is never `None`, then we can get rid of this